src/wp-includes/l10n.php

```diff
@@ -907,6 +932,8 @@ function load_plugin_textdomain( $domain, $deprecated = false, $plugin_rel_path
 		$path = WP_PLUGIN_DIR;
 	}
 
+	$wp_textdomain_registry->set( $domain, $locale, $path );
+
 	return load_textdomain( $domain, $path . '/' . $mofile );
```

Comment:
Shouldn't we pass the locale here since we have it?

```suggestion
	return load_textdomain( $domain, $path . '/' . $mofile, $locale );
```

Same for the other calls.